File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    var answer = 0;
    
    var map = Array(n).fill(1)
    
    for(const l of lost){
        map[l - 1] -= 1
    }
    
    for(const r of reserve){
        map[r - 1] += 1
    }
    
    for(let i = 0; i < n; i++){
        if(map[i] === 0){
            if(map[i - 1] > 1){
                map[i] += 1
                map[i - 1] -= 1
            }
            else if(map[i + 1] > 1){
                map[i] += 1
                map[i + 1] -= 1
            }
        }
    }
    
    for(let i = 0; i < n; i++){
        if(map[i] > 0) answer += 1
    }
    return answer;
}

File: 프로그래머스/1/42862. 체육복/체육복.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int[] clothes = new int[n + 1];
        Arrays.fill(clothes, 1);

        // lost -1
        for (int l : lost) clothes[l]--;

        // reserve +1
        for (int r : reserve) clothes[r]++;

        for (int i = 1; i <= n; i++) {
            if (clothes[i] == 0) {
                // 앞에서 빌리기
                if (i - 1 >= 1 && clothes[i - 1] == 2) {
                    clothes[i]++;
                    clothes[i - 1]--;
                }
                // 뒤에서 빌리기
                else if (i + 1 <= n && clothes[i + 1] == 2) {
                    clothes[i]++;
                    clothes[i + 1]--;
                }
            }
        }

        int answer = 0;

        for (int i = 1; i <= n; i++) {
            if (clothes[i] >= 1) answer++;
        }

        return answer;
    }
}